migrations/20210615130700-index-transaction-privacy-id.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+module.exports = {
+  up: async queryInterface => {
+    await queryInterface.sequelize.query(`
+      CREATE INDEX CONCURRENTLY privacy_transfer_id ON "Transactions" (((data ->> 'token')::text)) WHERE (data ->> 'token') IS NOT NULL;
```

Comment:
Since the index is created concurrently there's no risk of blocking the DB, this migration does a great job in this regard.

The only downside of the current approach is that Postgres needs to go through all the existing transactions, which could explain why it took so long. A simple condition on the date could have to speed it up:

```es6
WHERE (data ->> 'token') IS NOT NULL AND "createdAt" > '2021-03-01';
```

(I'm not sure when is the first privacy transaction)

Reply:
Thanks Ben! I'll keep that in mind for the next time!